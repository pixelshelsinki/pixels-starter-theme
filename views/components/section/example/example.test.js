import { render } from 'twig-testing-library'

describe('Example section', () => {
  test('Should render with all content', async () => {
    const { container } = await render(
      './views/components/section/example/example.twig',
      {
        section: {
          title: 'Title of section',
          content: 'Content of the seciton',
          button: {
            link: 'http://pixels.fi',
            title: 'Click me',
          },
        },
      },
      twigNamespaces,
    )

    expect(container).toMatchSnapshot()
  })

  test('Should render with partial content', async () => {
    const { container } = await render(
      './views/components/section/example/example.twig',
      {
        section: {
          title: 'Title of section',
        },
      },
      twigNamespaces,
    )

    expect(container).toMatchSnapshot()
  })
})
