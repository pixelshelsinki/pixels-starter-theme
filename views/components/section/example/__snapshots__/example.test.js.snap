// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example section Should render with all content 1`] = `
<div>
  <section
    class="section section- "
  >
    
    
    <div
      class="section-example__wrap"
    >
      
          
      <h3
        class="section-example__title"
      >
         Title of section 
      </h3>
      
    
          
      <p
        class="section-example__content"
      >
         Content of the seciton
      </p>
      
    
          
      <a
        class="btn"
        href="http://pixels.fi"
      >
        Click me
      </a>
      
    
  
    </div>
    

  </section>
  

</div>
`;

exports[`Example section Should render with partial content 1`] = `
<div>
  <section
    class="section section- "
  >
    
    
    <div
      class="section-example__wrap"
    >
      
          
      <h3
        class="section-example__title"
      >
         Title of section 
      </h3>
      
    
    
    
  
    </div>
    

  </section>
  

</div>
`;
